CMakeLists.txt

```diff
@@ -104,3 +92,56 @@ CONFIGURE_FILE(
 
 INSTALL(FILES "${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc"
         DESTINATION lib/pkgconfig)
+
+
+
+###############################################################################
+### testing
+if(BUILD_TESTS)
+    enable_testing()
+
+    # Download and unpack googletest at configure time
+    configure_file(CMakeLists.txt.in googletest/CMakeLists.txt)
```

Comment:
What is file `CMakeLists.txt.in` doing?

Reply:
I set this up according to the googletest documentation a while back, but I don't really know exactly.

Reply:
Ok, that's fine then.